e2e/stack_test.go

```diff
@@ -117,6 +124,44 @@ func TestE2E(t *testing.T) {
 	runningTests.Wait()
 }
 
+func checkForRollbacks(t *testing.T, stack *e2e.StackConfig) {
+	// Subscribe to new block events
+	const subscriber = "rollbackChecker"
+	eventChan, err := stack.L2Client.Subscribe(stack.Ctx, subscriber, bfttypes.QueryForEvent(bfttypes.EventNewBlock).String())
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, stack.L2Client.UnsubscribeAll(stack.Ctx, subscriber))
+	}()
+
+	var lastBlockHeight int64
+
+	// Check new block events for rollbacks
+	for event := range eventChan {
+		eventNewBlock, ok := event.Data.(bfttypes.EventDataNewBlock)
+		require.True(t, ok)
+		currentHeight := eventNewBlock.Block.Header.Height
+
+		// Skip the rollback check if lastBlockHeight is not initialized yet
+		if lastBlockHeight > 0 {
+			// Ensure that the current block height is the last checked height + 1
+			require.Equal(t, currentHeight, lastBlockHeight+1, "monomer has rolled back")
+		}
+		lastBlockHeight = currentHeight
+
+		// Get the L1 block info from the first tx in the block
+		ethTxs, err := monomer.AdaptCosmosTxsToEthTxs(eventNewBlock.Block.Txs)
+		require.NoError(t, err)
+		l1BlockInfo, err := derive.L1BlockInfoFromBytes(&rollup.Config{}, uint64(eventNewBlock.Block.Time.Unix()), ethTxs[0].Data())
```

Comment:
Might be best to just attach the full rollup config to the StackConfig now that I think about it. That will give us the SequencerWindowSize for free.